r/R/dplyr-functions.R

```diff
@@ -280,6 +284,81 @@ nse_funcs$str_trim <- function(string, side = c("both", "left", "right")) {
   Expression$create(trim_fun, string)
 }
 
+nse_funcs$substr <- function(string, start, stop) {
+  assert_that(
+    length(start) == 1,
+    msg = "`start` must be length 1 - other lengths are not supported in Arrow"
+  )
+  assert_that(
+    length(stop) == 1,
+    msg = "`stop` must be length 1 - other lengths are not supported in Arrow"
+  )
+
+  if (start <= 0) {
+    start <- 1
+  }
+
+  if (stop < start) {
+    stop <- 0
```

Comment:
Can you explain these? It's not obvious why this is correct, and the base R code and docs don't discuss these cases.